code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -974,9 +978,70 @@
 	if(prob(20))
 		M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 1*REM, 50)
 	M.adjustStaminaLoss(2.5*REM, 0)
+	mood.adjustPsychInstability(2)
 	..()
 	return TRUE
 
+/datum/reagent/medicine/haloperidol/on_mob_add(mob/living/L)
+	..()
+	ADD_TRAIT(L, TRAIT_UNDER_CONTROL, PSYCH_TRAIT)
+
+/datum/reagent/medicine/haloperidol/on_mob_end_metabolize(mob/living/L)
+	REMOVE_TRAIT(L, TRAIT_UNDER_CONTROL, PSYCH_TRAIT)
+
+/datum/reagent/medicine/haloperidol/overdose_process(mob/living/M)
+	var/datum/component/mood/mood = M.GetComponent(/datum/component/mood)
```

Comment:
every use of GetComponent is a violation of the encapsulation of the mood beahviour

Reply:
Why do we still have GetComponent?